library/psa_crypto.c

```diff
@@ -2011,12 +2000,27 @@ psa_status_t psa_import_key(const psa_key_attributes_t *attributes,
         }
     }
 
-    bits = slot->attr.bits;
-    status = psa_driver_wrapper_import_key(attributes,
-                                           data, data_length,
-                                           slot->key.data,
-                                           slot->key.bytes,
-                                           &slot->key.bytes, &bits);
+    if (PSA_KEY_TYPE_IS_ASYMMETRIC(attributes->core.type) &&
+        PSA_KEY_TYPE_IS_DH(attributes->core.type)) {
+        if (psa_is_dh_key_size_valid(PSA_BYTES_TO_BITS(data_length)) == 0) {
+            status = PSA_ERROR_INVALID_ARGUMENT;
+            goto exit;
+        }
+
+        /* Copy the key material. */
+        memcpy(slot->key.data, data, data_length);
+        bits = PSA_BYTES_TO_BITS(data_length);
+
+        status = PSA_SUCCESS;
```

Comment:
This comment is more about the design rather than the code itself, which looks ok.
While I agree with the changes you did in `psa_export_public_key()`, here it seems to me that you are managing FFDH as a "special case" compared to the other ones (RSA, ECC). The main reason seems to be that while RSA has `mbedtls_psa_rsa_import_key()` and ECC has `mbedtls_psa_ecp_import_key()`, DH misses its own `mbedtls_psa_rsa_import_key` (which I guess it would go inside a file named `psa_crypto_dh`).
For uniformity I would say it would be better to follow the same pattern. Wdyt?

PS: I think that the same change, if any, applies also to `psa_generate_derived_key_internal()` below.

Reply:
I'm also not sure about this as mentioned here https://github.com/Mbed-TLS/mbedtls/pull/7577#issuecomment-1543745482. 
We need to wait for input from @mpg .

Reply:
Ops, sorry, I didn't notice that comment. Yes, let's wait for Manuel's feedback ;)